### DISCUSSION TITLE
Feature request: Templates solutions x4

### DISCUSSION THREAD

**Discussion by flusteredfognozzle** [victorbalssa/abacus#180] [Q&A]
**Is your feature request related to a problem? Please describe.**
I am heavily using Abacus to track my expenses. There are several transactions that occur regularly and I am thus faced with performing repeated tasks to fill the same fields with the same values (except for e. g. the amount) over and over again.

**Describe the solution you'd like**
Solution 1: Templates (preferred solution)
Templates containing all of the fields of a transaction can be created by filling selected fields. By selecting "New transaction of template" a new transaction opens with the fields pre-filled according to the template. After making additions or changes (e. g. amount), the transaction is submitted.

Solution 2: Shortcuts integration
Allow for the triggering of pre-filled transactions from the iOS Shortcuts app.

Solution 3: URI schemes
Introduce extensive uri schemes so that pre-filled transactions can be triggered from outside of Abacus, e. g. by Launch.

Solution 4: Pre-filled fields based on the last transaction with the same title
When creating a new transaction in the current ui function the title field offers a selection of auto-completes. Upon selecting a title the form should be optionally pre-filled with the (exact) details of the last transaction with this title.

**Describe alternatives you've considered**
none

**Additional context**
none

**Comment by victorbalssa** [comment]
Hey @hessijames79,

Have you tried to duplicate your existing transaction by either long pressing one in the transaction list or sliding to the right and then click on `Clone`?

it will duplicate everything in a "New transaction" form with the current datetime, so you just need to update the `amount` field.

**Reply by flusteredfognozzle** [reply]
Hi!
No I haven't because my idea is to save time. Scrolling down the old transactions to find a matching one wouldn't be of much help.

Hessi

**Comment by victorbalssa** [comment]
Moving this to a discussion as it's a duplicate of #119
